src/transformers/models/siglip/convert_siglip_to_hf.py

```diff
@@ -285,15 +292,18 @@ def convert_siglip_checkpoint(model_name, pytorch_dump_folder_path, verify_logit
     # load HuggingFace model
     model = SiglipModel(config).eval()
     model.load_state_dict(state_dict)
-
     # create processor
     # important: make tokenizer not return attention_mask since original one doesn't require it
     image_size = config.vision_config.image_size
     size = {"height": image_size, "width": image_size}
     image_processor = SiglipImageProcessor(size=size)
     tokenizer = SiglipTokenizer(vocab_file=vocab_file, model_input_names=["input_ids"])
-    processor = SiglipProcessor(image_processor=image_processor, tokenizer=tokenizer)
 
+    if model_name == "siglip-so400m-patch14-224":
+        tokenizer.model_max_length = 16
```

Comment:
I don't think we set the `model_max_length` attribute for the existing checkpoints, should we do that? Would it break anything?

Reply:
for this one it is different, that's why I changed it